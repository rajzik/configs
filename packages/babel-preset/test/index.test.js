import { create } from 'babel-test';
import { toMatchFile } from 'jest-file-snapshot';
import path from 'node:path';

import { getConfig } from './test-utils';

const { fixtures } = create({ configFile: true });
const { test } = create({
  ...getConfig({ library: true, node: true, esm: false }),
  configFile: false,
  babelrc: false,
});

expect.extend({ toMatchFile });
fixtures('babel-config', path.join(__dirname, 'fixtures'));

describe('babel typescript test', () => {
  test('typescript transpilation', async ({ transform }) => {
    const { code } = await transform(
      `
      class Cls {
        #initialized = false;
        constructor() {
          this.#initialized = true
        }
      };
      export default Cls;`,
      { filename: 'index.ts' },
    );

    expect(code).toMatchSnapshot();
  });

  test('typescript decorators', async ({ transform }) => {
    const { code } = await transform(
      `
      class Cls {
        @decorator
        foo() {
          console.log('Test');
        }
      };
      export default Cls;`,
      { filename: 'index.ts' },
    );

    expect(code).toMatchSnapshot();
  });
});
