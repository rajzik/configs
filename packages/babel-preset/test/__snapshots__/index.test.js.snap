// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel typescript test typescript decorators 1`] = `
"\\"use strict\\";

var _interopRequireDefault = require(\\"@babel/runtime/helpers/interopRequireDefault\\");

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;

var _applyDecoratedDescriptor2 = _interopRequireDefault(require(\\"@babel/runtime/helpers/applyDecoratedDescriptor\\"));

var _class;

let Cls = (_class = class Cls {
  foo() {
    console.log('Test');
  }

}, ((0, _applyDecoratedDescriptor2.default)(_class.prototype, \\"foo\\", [decorator], Object.getOwnPropertyDescriptor(_class.prototype, \\"foo\\"), _class.prototype)), _class);
;
var _default = Cls;
exports.default = _default;"
`;

exports[`babel typescript test typescript transpilation 1`] = `
"\\"use strict\\";

var _interopRequireDefault = require(\\"@babel/runtime/helpers/interopRequireDefault\\");

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;

var _classPrivateFieldSet2 = _interopRequireDefault(require(\\"@babel/runtime/helpers/classPrivateFieldSet\\"));

function _classPrivateFieldInitSpec(obj, privateMap, value) { _checkPrivateRedeclaration(obj, privateMap); privateMap.set(obj, value); }

function _checkPrivateRedeclaration(obj, privateCollection) { if (privateCollection.has(obj)) { throw new TypeError(\\"Cannot initialize the same private elements twice on an object\\"); } }

var _initialized = /*#__PURE__*/new WeakMap();

class Cls {
  constructor() {
    _classPrivateFieldInitSpec(this, _initialized, {
      writable: true,
      value: false
    });

    (0, _classPrivateFieldSet2.default)(this, _initialized, true);
  }

}

;
var _default = Cls;
exports.default = _default;"
`;
